frontend/public/components/app.jsx

```diff
@@ -303,6 +305,24 @@ const CaptureTelemetry = React.memo(function CaptureTelemetry() {
   return null;
 });
 
+const SetTheme = React.memo(function SetTheme() {
```

Comment:
Have this component handle the updating of local storage whenever the theme changes via user preferences so that the user preference component can use the standard mechanism.